### ISSUE TITLE
Add option to make small increments the default

### ISSUE THREAD

**Issue by wmertens** [MonitorControl/MonitorControl#120]
My monitor gets way too loud way too soon, so it would be nice to map the mac volume to a slow-starting curve like `v = x/(2-x/100)` (where 0 <= x <= 100 is the mac volume)

**Comment by JoniVR** [comment]
To me this seems like a feature that adds a bit of complexity and only benefits a very small number of users, but that's just my opinion, I don't know how others feel about it.

**Comment by Nemo64** [comment]
I have a few LG monitors at work here and all have this Problem.
Every volume step increases the volume by 6 units on the display.
6 is fairly quiet, 12 is fairly loud but increasing the volume further bearly makes a difference up to a 100 (except that it distorts more)

But the shift + alt method works beautifully as a workaround in the mean time.

I would also be fine with just a lower maximum volume. If 50 would be the maximum that would already be louder than i'd ever use.

**Comment by silvetti** [comment]
I have the same issue, and agree with @Nemo64 Nemo, 6 is too low and 12 too high.
I have been working around it by using @JoniVR JoniVR suggestion but it would be great to have a more "default solution".

**Comment by inf0dr0id** [comment]
I've the same issue. A relatively quick fix could be an option to swap the default behavior of the volume keys. If turned on a volume key press would then behave as a shift+alt modified and vice versa.

**Comment by silvetti** [comment]
Yes, @inf0dr0id's idea is quite good!

**Comment by teilor** [comment]
Just want to add my +1 on this idea. The shift + alt workaround works fine for me, but would be nice to have an option to make it the default behaviour.

**Comment by phifogg** [comment]
Shift + Alt works as well hier (Philips screen) but having the option to change the increment would be awesome!

**Comment by xionc666** [comment]
It's sad that such simple change request is ignored. Basically changing:
`display.stepVolume(isUp: mediaKey == .volumeUp, isSmallIncrement: isSmallIncrement)`
to
`display.stepVolume(isUp: mediaKey == .volumeUp, isSmallIncrement: true)`
in AppDelegate.swift (row 280) would make the VOL keys to behave as the OPTION+SHIFT+VOL. Simple checkbox may be then added to control that line in the Preferences. 

Perhaps I'll commit such change, if I get the dev env working. I do build the MonitorControl in latest Xcode successfully, but it's not working as intended (without any changes in the code). The sliders in GUI are working, but the keys are changing the volume only once in several attempts. I'll try with Xcode v11.7...

**Comment by inf0dr0id** [comment]
My solution for this problem was switching to Lunar.
[https://github.com/alin23/Lunar](url)

**Comment by silvetti** [comment]
Isn’t Lunar just for monitor brightness?
Does it also work with volume keys?

Kind Regards,
Tomás Pita da Silva

**Comment by inf0dr0id** [comment]
Nope it does volume and contrast as well and you can set the steps for increasing and decreasing. I've set volume to 1 and now all is fine.

**Comment by silvetti** [comment]
Good to know!
Will give it a try as this is a very annoying problem (for me) with monitor control as now specially in Big Surr it does like 0-6-13
As in 6 too low and 13 way too loud...
Pity because apart from this issue I really don’t have any issues with monitor control.

Kind Regards,
Tomás Pita da Silva

**Comment by xionc666** [comment]
Well, controlling volume with Lunar doesn't work for me. I was however able to build the MonitorControl (xcode 11.3.1 required) with my simple change mentioned above and now it's working perfectly 👍 It's even better than Karabiner, bacause I have 100-step volume (+/-1) for external speakers and standard 16-step (+/-6) for the rest :)

**Comment by silvetti** [comment]
Hopefully MonitorControl devs will implement such feature (with maybe a checkbox to activate it in the Options).

**Comment by phifogg** [comment]
Thanks to @inf0dr0id. Lunar works flawlessly after mapping some of my media keys newly :D Perfect solution for me.

**Comment by waydabber** [comment]
This will be handled here in a generic fashion (it makes sense to add custom DDC min/max limits for range controlled functions - that will solve this issue as well):

https://github.com/MonitorControl/MonitorControl/issues/310

**Comment by Zulaaya** [comment]
I created these tools for import to Karabiner. It works well for version 14.3.

I hope you'll get a chance to see my response and that the tools can help you well!

Decrease Display Brightness by small increments using F1. Equivalent to using option + shift Long press, fast https://genesy.github.io/karabiner-complex-rules-generator/#eyJ0aXRsZSI6IkRlY3JlYXNlIERpc3BsYXkgQnJpZ2h0bmVzcyBieSBzbWFsbCBpbmNyZW1lbnRzIHVzaW5nIEYxLiBFcXVpdmFsZW50IHRvIHVzaW5nIG9wdGlvbiArIHNoaWZ0IiwicnVsZXMiOlt7ImRlc2NyaXB0aW9uIjoiRGVjcmVhc2UgRGlzcGxheSBCcmlnaHRuZXNzIGJ5IHNtYWxsIGluY3JlbWVudHMgdXNpbmcgRjEuIExvbmcgcHJlc3MgZm9yIGZhc3QgZGVjcmVhc2UuIiwibWFuaXB1bGF0b3JzIjpbeyJ0eXBlIjoiYmFzaWMiLCJmcm9tIjp7ImtleV9jb2RlIjoiZjEifSwidG8iOlt7InJlcGVhdCI6dHJ1ZSwia2V5X2NvZGUiOiJkaXNwbGF5X2JyaWdodG5lc3NfZGVjcmVtZW50IiwibW9kaWZpZXJzIjpbImxlZnRfc2hpZnQiLCJsZWZ0X2FsdCJdfV0sInRvX2lmX2hlbGRfZG93biI6W3sicmVwZWF0Ijp0cnVlLCJrZXlfY29kZSI6ImRpc3BsYXlfYnJpZ2h0bmVzc19kZWNyZW1lbnQifV19XX1dfQ==


Increase Display Brightness by small increments using F2. Equivalent to using option + shift Long press, fast
https://genesy.github.io/karabiner-complex-rules-generator/#eyJ0aXRsZSI6IkluY3JlYXNlIERpc3BsYXkgQnJpZ2h0bmVzcyBieSBzbWFsbCBpbmNyZW1lbnRzIHVzaW5nIEYyLiBFcXVpdmFsZW50IHRvIHVzaW5nIG9wdGlvbiArIHNoaWZ0IiwicnVsZXMiOlt7ImRlc2NyaXB0aW9uIjoiSW5jcmVhc2UgRGlzcGxheSBCcmlnaHRuZXNzIGJ5IHNtYWxsIGluY3JlbWVudHMgdXNpbmcgRjIuIExvbmcgcHJlc3MgZm9yIGZhc3QgaW5jcmVhc2UuIiwibWFuaXB1bGF0b3JzIjpbeyJ0eXBlIjoiYmFzaWMiLCJmcm9tIjp7ImtleV9jb2RlIjoiZjIifSwidG8iOlt7InJlcGVhdCI6dHJ1ZSwia2V5X2NvZGUiOiJkaXNwbGF5X2JyaWdodG5lc3NfaW5jcmVtZW50IiwibW9kaWZpZXJzIjpbImxlZnRfc2hpZnQiLCJsZWZ0X2FsdCJdfV0sInRvX2lmX2hlbGRfZG93biI6W3sicmVwZWF0Ijp0cnVlLCJrZXlfY29kZSI6ImRpc3BsYXlfYnJpZ2h0bmVzc19pbmNyZW1lbnQifV19XX1dfQ==
